### PR TITLE
Fix sorting and add test

### PR DESCRIPTION
**Pull request by fareed-ah**
Dear fellow team members,

This PR contains the following additions to our wonderful project:
- Fixed sorting the percs
- Add another test case to ensure sorting the percs is done properly

Muy bien,
Fareed

### CODE REVIEW

**Comment by MaximeNdutiye** [comment]
it looks like you're not up to date. can you rebase your branch with master please?
image

File: src/test/java/com/perkmanager/integration/PerksControllerTest.java [hunk 1]
```diff
@@ -191,4 +191,48 @@ public void canSearchByFilterAndSort() throws Exception {
...
+        Perk p2 = new Perk(2, "product2", "description2");
+        p2.upVote();
+        Membership m2 = new Membership("title2", "type");
+        p2.setMembership(m2);
+
+        perkRepository.save(p1);
+        perkRepository.save(p2);
+
+        // given
+        given(perkRepository.findAll(any(Specification.class), any(Sort.class))).
```
Comment (MaximeNdutiye) [inline src/test/java/com/perkmanager/integration/PerksControllerTest.java:210]: maybe we could  remove this stub so we actually test the repository as well?

Comment (fareed-ah) [reply src/test/java/com/perkmanager/integration/PerksControllerTest.java:205]: Nope I don't think it works like that big man

File: src/test/java/com/perkmanager/integration/PerksControllerTest.java [hunk 2]
```diff
@@ -186,4 +186,50 @@ public void canSearchByFilterAndSort() throws Exception {
...
+                willReturn(Arrays.asList(p2,p1));
+
+        // when
+        MockHttpServletResponse response = mvc.perform(
+                get("/perks/search?value=''&sortBy=1").contentType(MediaType.APPLICATION_JSON)
+        ).andReturn().getResponse();
+
+        // then
+        assertThat(response.getStatus()).isEqualTo(HttpStatus.OK.value());
+        assertThat(response.getContentAsString()).isEqualTo(
```
Comment (zachary-nguyen) [inline src/test/java/com/perkmanager/integration/PerksControllerTest.java:215]: Nice, checking that list is in order

**Review by zachary-nguyen — APPROVED** [review]

**Review by MaximeNdutiye — APPROVED** [review]

**Pull request merged**